sources/cassandra/README.md

```diff
@@ -0,0 +1,228 @@
+   - Establish network connectivity
+
+2. **Application Configuration**
+   - Point application to ZDM proxy
+   - Verify dual-write functionality on Cassandra and Spanner
+
+3. **Data Migration**
+   - Execute [bulk data migration](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/sourcedb-to-spanner/README_Sourcedb_to_Spanner.md#cassandra-to-spanner-bulk-migration)
+   - Monitor progress and performance
+
+4. **Validation**
+   - Perform data validation over row counts and row data
+   - Reconcile any inconsistencies
+   - Refer to the [validation scripts](./validations/) for validating data consistency between source and target clusters.
+
+5. **Cutover**
+   - Switch application to direct Spanner connection
+   - Decommission proxies
+
+## Prerequisites
+
+- Docker (for development/testing)
+- Terraform (for production deployment)
+- Google Cloud Platform project with Spanner instance
+- Service account with appropriate permissions
+- Network connectivity between components
+
+## Getting Started
+
+### Single Instance Testing Setup
+
+This guide helps you set up a single instance of the proxy for development or testing purposes using Docker.
+
+#### Prerequisites
+
+Before you begin, ensure you have:
+- Docker installed on your system
+- Network connectivity between:
+  - Your VM/local machine
+  - Source Cassandra cluster
+  - Target Spanner instance
+  - Your application
+- A Google Cloud Service Account key file with permissions to:
+  - Write to Spanner instance
+  - Create and manage databases
+- A Spanner instance and database already set up
+- The Spanner database name matches your Cassandra keyspace name
+
+#### Step 1: Build the image
+
+- Keep the entrypoint.sh in the same directory as the Dockerfile  
+
+- Run the following command to build the local image:
+
+```bash
+docker build -t zdm-proxy:latest .
+```
+
+#### Step 2: Run the container
+1. Ensure your service account key file (e.g., `keys.json`) is in the directory
+
+2. Create a `zdm-config.yaml` file with your configuration. See [sample-zdm-config.yaml](./sample-zdm-config.yaml) for an example.
+
+3.  Run the container with the following command:
+
+```bash
+docker run -d -p 14002:14002 \
+-v $(pwd)/zdm-config.yaml:/zdm-config.yaml \
+-v $(pwd)/keys.json:/var/run/secret/keys.json \
+-e SPANNER_PROJECT=your-project-id \
+-e SPANNER_INSTANCE=your-instance-id \
+-e SPANNER_DATABASE=your-database-id \
+-e GOOGLE_APPLICATION_CREDENTIALS="/var/run/secret/keys.json" \
+-e ZDM_CONFIG=/zdm-config.yaml \
+zdm-proxy:latest
+```
+
+Replace the following values:
+- `your-project-id`: Your Google Cloud project ID
+- `your-instance-id`: Your Spanner instance ID
+- `your-database-id`: Your Spanner database ID
+
+#### Step 4: Verify the Setup
+
+1. Check if the container is running:
+```bash
+docker ps
+```
+
+2. View the container logs:
+```bash
+docker logs <container-id>
+```
+
+3. Test the connection using cqlsh:
+```bash
+cqlsh localhost 14002
```

Comment:
I thought the default port for cassandra is 9042. Is this the port for ZDM ?

Reply:
Correct